### DISCUSSION TITLE
Need help using deleteObjects with presignedurl

### DISCUSSION THREAD

**Discussion by RProduction** [aws/aws-sdk-js-v3#5346] [Q&A]
Hello, I need help with using deleteObjects using presignedurl.
Is it possible to do this? I need it for usage in edge runtime.
I can use deleteObject with presigned url but with deleteObjects it resulted in invalid key.
I already tried using it with delete method and post method but the result is the same.

If I use deleteObjects without presignedurl it will result in undefined filereader error.
Please give me solution for this problem.

**Comment by yenfryherrerafeliz** [comment]
Hi @RProduction, could you please provide the following so I can better assist you:
- SDK minor version.
- Sample code that you have tried with.

Thanks!

**Reply by RProduction** [reply]
@aws-sdk/client-s3: "^3.423.0"
@aws-sdk/s3-request-presigner: "^3.427.0"

Sample code:
`
export async function DeleteFile(id: string) {
  const deleteCommand = new DeleteObjectCommand({
    Bucket: bucket,
    Key: id,
  })

  const url = await getSignedUrl(s3Client, deleteCommand);
  await fetch(url, {
    method: 'DELETE'
  });
}

export async function DeleteFiles(ids: string[]) {
  const objs = ids.map((value) => ({Key: value}))

  const deleteCommand = new DeleteObjectsCommand({
    Bucket: bucket,
    Delete: {
      Objects: objs
    }
  })

  const res = await s3Client.send(deleteCommand);
}
`

DeleteFile is function using presigned url to delete one object and it is working.
DeleteFiles is function without presigned url that work on nodejs runtime of nextjs but not on edge runtime.

**Reply by RProduction** [reply]
Any solution to my problem? I really need to use this in edge runtime
